model/flow/dkg.go

```diff
@@ -31,51 +31,64 @@ func (state DKGEndState) String() string {
 	}
 }
 
-// DKGIndexMap describes the membership of the DKG committee 𝒟. Flow's random beacon utilizes
-// a threshold signature scheme, which requires a Distributed Key Generation [DKG] to generate the
-// key shares for each committee member. In the formal cryptographic protocol for DKG with n parties,
-// the individual participants are solely identified by indices {0, 1, ..., n-1} and the fact that these
-// are non-negative integer values is actively used by the DKG protocol. Accordingly, our implementation
-// of the lower-level cryptographic primitives work with these DKG index values.
+// DKGIndexMap completely describes the DKG committee 𝒟 of size |𝒟| = n.
+// Flow's random beacon utilizes a threshold signature scheme run by the committee 𝒟.
+// In the formal cryptographic protocol for a threshold signature with n parties,
+// the individual participants are identified by n public distinct non-negative integers, or simply indices.
+// These public indices are agreed upon by all participants and are used by the low-level
+// Shamir Secret Sharing [SSS].
```

Comment:
Sorry, I should have mentioned that I was simply trying to streamline the presentation. I am assuming that most engineers will only read the first paragraph and the more concise that is the better.

* I think mentioning in the introduction our convention that the index set must resemble `{0, 1, ..., n-1}` is important, because it helps with understanding how to use `DKGIndexMap` and understanding the following details in the documentation (for engineers that actually continue reading beyond the first paragraph). So I am suggesting that we only say "this is the convention we have (without loss of generality)".
* I would suggest to move the explanation of "why" this convention exists further down". We could put it [here](https://github.com/onflow/flow-go/blob/f8a974947d92a3148bea4072e7679febaa5ce433/model/flow/dkg.go#L56-L57) or [here](https://github.com/onflow/flow-go/blob/f8a974947d92a3148bea4072e7679febaa5ce433/model/flow/dkg.go#L67-L69) maybe?

Reply:
> I think mentioning in the introduction our convention that the index set must resemble {0, 1, ..., n-1} is important

I see your point. In that case I suggest to move the [formal specification](https://github.com/onflow/flow-go/blob/f8a974947d92a3148bea4072e7679febaa5ce433/model/flow/dkg.go#L55-L59) section up to the very beginning since it is the most important to developers and engineers. Curious engineers can continue further but engineers who stop reading early get the essential implementation convention point. 

I added the change [here](https://github.com/onflow/flow-go/pull/6590/commits/b0f36be37db031fd8b43a5e929b360c6c7c13fab) but happy to reorganize again.

Reply:
Amazing! 1 sentence context, then formal specification of what has to hold. Only then all the context. Love it.